### PR TITLE
[Matrix] Remove legacy code of matrix implementation

### PR DESCRIPTION
**Pull request by yubingex007-a11y**

### CODE REVIEW

File: llvm-spirv/test/extensions/INTEL/SPV_INTEL_joint_matrix/array_of_matrices.ll [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ;; Compiled from joint_matrix_bf16_fill_k_cache.cpp from https://github.com/intel/llvm
-;; command: clang++ -fsycl -DSYCL_EXT_ONEAPI_MATRIX_VERSION=4 llvm/sycl/test-e2e/Matrix/joint_matrix_bf16_fill_k_cache.cpp -fsycl-device-only -o test.bc
```
Comment (yubingex007-a11y) [inline llvm-spirv/test/extensions/INTEL/SPV_INTEL_joint_matrix/array_of_matrices.ll:2]: @MrSidims i guess we need modify the file in kronos code, right?

**Comment by dkhaldi** [comment]
@JackAKirk, we are remove the macro SYCL_EXT_ONEAPI_MATRIX_VERSION as we maintain only one version of the API implementation. Can you please make sure CUDA and AMD implementations are not affected (as they only use SYCL_EXT_ONEAPI_MATRIX_VERSION=4 which is now the only one supported).

**Comment by JackAKirk** [comment]
> Can you please make sure CUDA and AMD implementations are not affected (as they only use SYCL_EXT_ONEAPI_MATRIX_VERSION=4 which is now the only one supported)?

Yes they only use SYCL_EXT_ONEAPI_MATRIX_VERSION=4, so it is fine to remove the macro.

**Review by JackAKirk — APPROVED** [review]
LGTM from cuda/hip perspective.

File: llvm-spirv/test/extensions/INTEL/SPV_INTEL_joint_matrix/array_of_matrices.ll (hunk 1 above)
Comment (steffenlarsen) [reply llvm-spirv/test/extensions/INTEL/SPV_INTEL_joint_matrix/array_of_matrices.ll:2]: Yes, please revert the changes here and make them [upstream](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/tree/main). We can move ahead with the intel/llvm changes with or without the removal of these in upstream, right?

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Review by YuriPlyakhin — APPROVED** [review]

**Review by gmlueck — APPROVED** [review]

**Comment by againull** [comment]
@yubingex007-a11y Redhat build is failing in pre-commit.

**Comment by yubingex007-a11y** [comment]
> @yubingex007-a11y Redhat build is failing in pre-commit. Could you please take a look.

yep, it can be reproduced by gcc7.5 locally. i guess it is because we remove "#include <sycl/detail/defines.hpp> // for SYCL_EXT_ONEAPI_MATRIX_VERSION" in spirv_type.hpp

**Review by jchlanda — APPROVED** [review]
CUDA/HIP 👍

**Comment by steffenlarsen** [comment]
Windows failures seem to be infrastructural issues. I would prefer we try to rerun them. I have retriggered a rerun.

**Pull request merged**